src/EPR.Calculator.Frontend.UnitTests/LocalAuthorityDisposalCostsControllerTests.cs

```diff
@@ -1,19 +1,212 @@
+        [ExpectedException(typeof(UriFormatException))]
+        public async Task GetHttpRequest_NullOrEmptyLapcapRunApi_ThrowsArgumentNullException()
+        {
+            // Arrange
+            var inMemorySettings = new Dictionary<string, string>
+            {
+                { $"{ConfigSection.LapcapSettings}:{ConfigSection.LapcapSettingsApi}", " " }
+            };
+
+            IConfiguration configuration = new ConfigurationBuilder()
+                .AddInMemoryCollection(inMemorySettings)
+                .Build();
+
+            var clientFactory = new HttpClientFactoryStub();
+
+            // Act
+            var result = await LocalAuthorityDisposalCostsController.GetHttpRequest(configuration, clientFactory);
+
+            // Assert is handled by ExpectedException
+        }
+
+        [TestMethod]
+        public async Task Index_SuccessfulResponse_ReturnsViewWithGroupedData()
+        {
+            var mockHttpMessageHandler = new Mock<HttpMessageHandler>();
+            mockHttpMessageHandler
+                   .Protected()
+                .Setup<Task<HttpResponseMessage>>(
+                    "SendAsync",
+                    ItExpr.IsAny<HttpRequestMessage>(),
+                    ItExpr.IsAny<CancellationToken>())
+                .ReturnsAsync(new HttpResponseMessage
+                {
+                    StatusCode = HttpStatusCode.OK,
+                    Content = new StringContent(JsonConvert.SerializeObject(MockData.GetLocalAuthorityDisposalCosts()))
+                });
+
+            var httpClient = new HttpClient(mockHttpMessageHandler.Object);
+
+            // Mock IHttpClientFactory
+            var mockHttpClientFactory = new Mock<IHttpClientFactory>();
+            mockHttpClientFactory
+                .Setup(_ => _.CreateClient(It.IsAny<string>()))
+                .Returns(httpClient);
+
+            // Arrange
+            var controller = new LocalAuthorityDisposalCostsController(GetConfigurationValues(), mockHttpClientFactory.Object);
+
+            // Act
+
             var result = controller.Index() as ViewResult;
+
+            // Assert
             Assert.IsNotNull(result);
             Assert.AreEqual(ViewNames.LocalAuthorityDisposalCostsIndex, result.ViewName);
+            Assert.IsInstanceOfType(result.Model, typeof(List<IGrouping<string, LocalAuthorityViewModel>>));
+        }
+
+        [TestMethod]
+        public void Index_WhenExceptionThrown_RedirectsToErrorPage()
+        {
+            // Arrange
+            var controller = new LocalAuthorityDisposalCostsController(null, null);
+
+            // Act
+            var result = controller.Index() as RedirectToActionResult;
+
+            // Assert
+            Assert.AreEqual("Index", result.ActionName);
+            Assert.AreEqual("StandardError", result.ControllerName);
+        }
+
+        private static IConfiguration GetConfigurationValues()
+        {
+            var builder = new ConfigurationBuilder()
```

Comment:
This implementation can be reused for all the unit tests. Can you move it to a TestHelper and replace the old Configuration items? Thanks

Reply:
sure

Reply:
This is optional comment. We could do it later

Reply:
done